src/overlays/actors/ovl_En_Go2/z_en_go2.c

```diff
@@ -30,199 +216,2028 @@ const ActorInit En_Go2_InitVars = {
+            }
+        case 1:
+            break;
+        default:
+            return;
+    }
+
+    if (func_8010BDBC(&globalCtx->msgCtx) != 2) {
+        player->actor.freezeTimer = 10;
+    } else {
+        gSaveContext.infTable[16] |= 0x1000;
+        this->unk_26E = 1;
+        this->unk_211 = false;
+        this->isAwake = false;
+        this->actionFunc = EnGo2_CurledUp;
+    }
+}
+
+void EnGo2_GoronFireGenericAction(EnGo2* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    Vec3s D_80A4854C = { 0x00, 0x00, 0x00 };
+
+    switch (this->goronState) {
+        // Wake up???
+        case 0:
+            if (func_8010BDBC(&globalCtx->msgCtx) == 2) {
+                EnGo2_GoronFireCamera(this, globalCtx);
+                globalCtx->msgCtx.msgMode = 0x37;
+                func_80034EC0(&this->skelAnime, sAnimations, 2);
+                this->waypoint = 1;
+                this->skelAnime.animPlaybackSpeed = 2.0f;
+                func_80A44D84(this);
+
+                this->actor.shape.rot = this->actor.posRot.rot;
+                this->animTimer = 60;
+                this->actor.gravity = 0.0f;
+                this->actor.speedXZ = 2.0f;
+                this->unk_194.unk_08 = D_80A4854C;
+                this->unk_194.unk_0E = D_80A4854C;
+                this->goronState++;
+                this->goronState++;
+                player->actor.posRot.rot.y = this->actor.posRot.rot.y;
+                player->actor.shape.rot.y = this->actor.posRot.rot.y;
+                player->actor.posRot.pos.x =
+                    (f32)((Math_Sins(this->actor.posRot.rot.y) * -30.0f) + this->actor.posRot.pos.x);
+                player->actor.posRot.pos.z =
+                    (f32)((Math_Coss(this->actor.posRot.rot.y) * -30.0f) + this->actor.posRot.pos.z);
+                func_8002DF54(globalCtx, &this->actor, 8);
+                func_800F5C64(0x51);
+            }
+            break;
+        // Walking away
+        case 2:
+            if (DECR(this->animTimer)) {
+                if (!(this->animTimer % 8)) {
+                    Audio_PlayActorSound2(&this->actor, NA_SE_EN_MORIBLIN_WALK);
+                }
+                Actor_MoveForward(&this->actor);
+            } else {
+                this->animTimer = 0;
+                this->actor.speedXZ = 0.0f;
+                if ((((this->actor.params & 0xFC00) >> 0xA) != 1) && (((this->actor.params & 0xFC00) >> 0xA) != 2) &&
+                    (((this->actor.params & 0xFC00) >> 0xA) != 4) && (((this->actor.params & 0xFC00) >> 0xA) != 5) &&
+                    (((this->actor.params & 0xFC00) >> 0xA) != 9) && (((this->actor.params & 0xFC00) >> 0xA) != 11)) {
+                    this->goronState++;
+                }
+                this->goronState++;
+            }
+            break;
+        // Walking away
+        case 3:
+            this->animTimer++;
+
+            if (!(this->animTimer % 8) && (this->animTimer < 10)) {
```

Comment:
Same as previous comment
```suggestion
            if (((this->animTimer % 8) == 0) && (this->animTimer < 10)) {
```

Reply:
Same as above, only matches with the negation `!`